### PR TITLE
Added lesson controller, views

### PR DESCRIPTION
**Pull request by rofiuddin**

### CODE REVIEW

File: app/controllers/lessons_controller.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
+class LessonsController < ApplicationController
+  before_action :logged_in_user
+
+  def index
+  @lessons = current_user.lessons
```
Comment (chuanhtuan) [inline app/controllers/lessons_controller.rb]: indent

File: app/controllers/lessons_controller.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  end
+
+  def show
+    @lesson = Lesson.find params[:id]
+  end
+
+  def create
+    category = Category.find params[:lesson][:category_id]
+    @lesson = category.lessons.create user_id: current_user.id
+    generate_results @lesson # generate 20 unique results
```
Comment (chuanhtuan) [inline app/controllers/lessons_controller.rb]: use call back function: before_create in Lesson model

File: app/controllers/lessons_controller.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    redirect_to @lesson
+
```
Comment (chuanhtuan) [inline app/controllers/lessons_controller.rb]: remove empty line

File: app/controllers/lessons_controller.rb [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  end
+
+  def update
+    @lesson = Lesson.find params[:id]
+    if @lesson.update_attributes params_lesson
+
```
Comment (chuanhtuan) [inline app/controllers/lessons_controller.rb]: remove empty line

File: app/models/lesson.rb [hunk 5]
```diff
@@ -1,2 +1,18 @@
...
+  has_many :words, through: :results
+
+  accepts_nested_attributes_for :results
+
+  before_save :update_result
+  validates :progress,  presence: true, numericality: { greater_than_or_equal_to: 0 }
+
+  private
+  def update_result
+  self.result = results.select do
```
Comment (chuanhtuan) [inline app/models/lesson.rb]: indent

File: app/views/lessons/_lesson.html.erb [hunk 6]
```diff
@@ -0,0 +1,10 @@
+<li class="lesson">
+  <span class="user"><%= link_to lesson.user.username, lesson.user %></span>
+  <span class="content">
+    Learn <%= pluralize lesson.results.answer_not_nil.count, "word" %>
```
Comment (chuanhtuan) [inline app/views/lessons/_lesson.html.erb]: i18n

File: app/views/lessons/_lesson_link.html.erb [hunk 7]
```diff
@@ -0,0 +1,4 @@
+<div>
+  <h4><%= "Lesson #{lesson.id}" %></h4>
```
Comment (chuanhtuan) [inline app/views/lessons/_lesson_link.html.erb]: i18n

File: app/views/users/show.html.erb [hunk 8]
```diff
@@ -11,12 +11,11 @@
...
     <%= render 'follow_form' if logged_in? %>
-    <% if @user.activities.any? %>
-      <h3>Activities (<%= @user.activities.count %>)</h3>
-      <ol class="activities">
-        <%= render @activities %>
-      </ol>
-      <%= will_paginate @activities %>
+    <% if @user.lessons.any? %>
+    <div class="activity">
+      <h3>Activities</h3>
```
Comment (chuanhtuan) [inline app/views/users/show.html.erb]: i18n

File: app/models/lesson.rb (hunk 5 above)
Comment (linhnt) [inline app/models/lesson.rb]: do |result|
end

File: app/models/lesson.rb [hunk 9]
```diff
@@ -1,2 +1,24 @@
 class Lesson < ActiveRecord::Base
+  belongs_to :user
+  belongs_to :category
+  has_many :results, dependent: :destroy
+  has_many :words, through: :results
+
+  accepts_nested_attributes_for :results
+  before_create: create_results
+  before_save :update_result
```
Comment (chuanhtuan) [inline app/models/lesson.rb]: before_update

File: app/models/lesson.rb [hunk 10]
```diff
@@ -1,2 +1,24 @@
...
+  validates :progress,  presence: true, numericality: { greater_than_or_equal_to: 0 }
+
+  private
+  def create_results
+    @unique_word = lesson.category.words.order("RAND()").limit 20
+    @unique_word.each do |word|
+      lesson.results.create word_id: word.id
```
Comment (chuanhtuan) [inline app/models/lesson.rb]: lesson.results.build word_id: word.id
because you use before_create

**Pull request merged**